test/controllers/nonsipp/memberpayments/UnallocatedContributionCYAControllerSpec.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp.memberpayments
+
+import config.Refined.OneTo5000
+import controllers.ControllerBaseSpec
+import eu.timepit.refined.refineMV
+import models.CheckOrChange
+import pages.nonsipp.memberpayments.UnallocatedEmployerAmountPage
+import play.api.inject.bind
+import play.api.inject.guice.GuiceableModule
+import services.PsrSubmissionService
+import views.html.CYAWithRemove
+
+class UnallocatedContributionCYAControllerSpec extends ControllerBaseSpec {
+
+  private implicit val mockPsrSubmissionService: PsrSubmissionService = mock[PsrSubmissionService]
+
+  override protected val additionalBindings: List[GuiceableModule] = List(
+    bind[PsrSubmissionService].toInstance(mockPsrSubmissionService)
+  )
+
+  override protected def beforeAll(): Unit =
+    reset(mockPsrSubmissionService)
+
+  private val index = refineMV[OneTo5000](1)
+
+  private def onPageLoad(checkOrChange: CheckOrChange) =
+    routes.UnallocatedContributionCYAController.onPageLoad(srn, checkOrChange)
+
+  private def onSubmit(checkOrChange: CheckOrChange) =
+    routes.UnallocatedContributionCYAController.onSubmit(srn, checkOrChange)
+
+  private val filledUserAnswers = defaultUserAnswers
+    .unsafeSet(UnallocatedEmployerAmountPage(srn), money)
+
+  "MoneyBorrowedCYAController" - {
+
+    List(CheckOrChange.Check, CheckOrChange.Change).foreach { checkOrChange =>
+      act.like(
+        renderView(onPageLoad(checkOrChange), filledUserAnswers) { implicit app => implicit request =>
+          injected[CYAWithRemove].apply(
+            UnallocatedContributionCYAController.viewModel(
+              ViewModelParameters(
+                srn,
+                schemeName,
+                money,
+                checkOrChange
+              )
+            )
+          )
+        }.withName(s"render correct ${checkOrChange.name} view")
+      )
+
+      act.like(
+        redirectNextPage(onSubmit(checkOrChange))
+          .before(MockPSRSubmissionService.submitPsrDetails())
```

Comment:
Could you please consider adding a check to see if the back end was called here? E.g. 
````
.after(
          verify(mockPsrSubmissionService, times(1)).submitPsrDetails(any())(any(), any(), any())
        )